oaknational/Oak-Web-Application
# Style/login

## Description

- adds style lint rule
- updates sign-in screens to use styled components
- replaces `window.prompt` with an input in the case that the user clicks the magic link on another device
  - updates tests to reflect this 
- uses `useStableCallback` for auth functions to ensure they're only called once (previously these functions were being called excessively).
- updates `IconButton` component to match `Button` signature
- moves `SignIn` component into its own file


## How to test

1. Go to {cloud link}
2. Go through log in flow
3. You should be able to log in without error
